lib/matplotlib/pyplot.py

```diff
@@ -138,17 +143,35 @@
 _log = logging.getLogger(__name__)
 
 
-def _copy_docstring_and_deprecators(method, func=None):
+@overload
+def _copy_docstring_and_deprecators(
+    method: Any,
+    func: Literal[None] = None
+) -> Callable[[Callable[_P, _R]], Callable[_P, _R]]: ...
+
+
+@overload
+def _copy_docstring_and_deprecators(
+    method: Any, func: Callable[_P, _R]) -> Callable[_P, _R]: ...
+
+
+def _copy_docstring_and_deprecators(
+    method: Any,
```

Comment:
I think this is fine, there are ways of specifying, e.g. something like:

```python
class HasDocstring(typing.Protocol):
   __doc__: str
```

but that seems perhaps overkill, especially for private functionality (also, haven't actually tested that... it could behave oddly for builtin behaviors like `__doc__`)